packages/nx/src/daemon/cache.ts

```diff
@@ -35,7 +30,9 @@ export function deleteDaemonJsonProcessCache(): void {
 export async function writeDaemonJsonProcessCache(
   daemonJson: DaemonProcessJson
 ): Promise<void> {
-  await writeJson(serverProcessJsonPath, daemonJson);
+  // TODO: Ask about it in the PR.
+  // `fs-extra` appends an extra newline at the end when writing JSON. Do we want it here?
+  await writeFile(serverProcessJsonPath, JSON.stringify(daemonJson));
```

Comment:
As per the comment. What do you think?

Reply:
We have our own `writeJsonFile` and `readJsonFile` utils that are already built upon `node:fs` in `packages/nx/src/utils/fileutils.ts`, please could you use those instead? You can set the appendNewLine option to make it match the previous behavior

Reply:
Changed per the request, but those utils are synchronous. Is this fine for you or should I add similar ones, but asynchronous?